website/templates/project/addon/widget.mako

```diff
@@ -1,31 +1,33 @@
-<div class="panel panel-default" name="${short_name}">
+% if complete or 'write' in user['permissions']:
+    <div class="panel panel-default" name="${short_name}">
 
-            <div class="panel-heading clearfix">
-                <h3 class="panel-title">${full_name}</h3>
-                <div class="pull-right">
-                    % if has_page:
-                       <a href="${node['url']}${short_name}/">  <i class="fa fa-external-link"></i> </a>
-                   % endif
+                <div class="panel-heading clearfix">
+                    <h3 class="panel-title">${full_name}</h3>
+                    <div class="pull-right">
+                        % if has_page:
+                           <a href="${node['url']}${short_name}/">  <i class="fa fa-external-link"></i> </a>
+                       % endif
 
+                    </div>
                 </div>
-            </div>
 
-    % if complete:
+        % if complete:
```

Comment:
Yeah—the entire thing hides if it's incomplete and the viewer is not write-enabled, but if the user is write-enabled _or_ the widget is complete, then it will show. Then, whether or not the panel's contents are shown depend on the completion status.
![diagram](https://cloud.githubusercontent.com/assets/5885378/16817890/ec5db79c-4912-11e6-9267-2f7743511dc2.png)